src/ripple/app/tx/impl/Transactor.h

```diff
@@ -51,6 +51,66 @@ struct PreflightContext
     }
 };
 
+struct PreflightResult
+{
+public:
+    PreflightContext const ctx;
+    TER const ter;
+
+    PreflightResult(PreflightContext const& ctx_,
+        TER ter_)
+        : ctx(ctx_)
+        , ter(ter_)
+    {
+    }
+};
+
+/** State information when determining if a tx is likely to claim a fee. */
+struct PreclaimContext
+{
+public:
+    Application& app;
+    ReadView const& view;
+    TER const preflightResult;
+    STTx const& tx;
+    ApplyFlags flags;
+    beast::Journal j;
+
+    PreclaimContext(Application& app_, ReadView const& view_,
+        TER preflightResult_, STTx const& tx_,
+            ApplyFlags flags_, beast::Journal j_ = {})
+        : app(app_)
+        , view(view_)
+        , preflightResult(preflightResult_)
+        , tx(tx_)
+        , flags(flags_)
+        , j(j_)
+    {
+    }
+};
+
+struct PreclaimResult
+{
+public:
+    PreclaimContext const ctx;
```

Comment:
Mainly as a shortcut to avoid having to copy all the individual members of the context into the result.